test/pipelinerun_test.go

```diff
@@ -160,14 +161,36 @@ func TestPipelineRun(t *testing.T) {
 				if !r.Status.GetCondition(duckv1alpha1.ConditionSucceeded).IsTrue() {
 					t.Fatalf("Expected TaskRun %s to have succeeded but Status is %v", taskRunName, r.Status)
 				}
-				for name, key := range map[string]string{
-					pipelineRunName: pipeline.PipelineRunLabelKey,
-					pipelineName:    pipeline.PipelineLabelKey,
-				} {
-					expectedName := getName(name, i)
-					lbl := pipeline.GroupName + key
-					if val := r.Labels[lbl]; val != expectedName {
-						t.Errorf("Expected label %s=%s but got %s", lbl, expectedName, val)
+
+				// Check label propagation to Tasks
+				labels := make(map[string]string, len(pr.ObjectMeta.Labels)+3)
```

Comment:
why +3 here? (maybe explain with a comment or a `const` ?)

Reply:
`len(pr.ObjectMeta.Labels)` for custom labels, `+2` for the static pipeline/pipelineRun label from the PipelineRun controller, and `+1` for the taskRun label from the `TaskRun` controller. I'll add a comment 😄